src/components/date-picker/flatpickr-adapter/flatpickr-adapter.tsx

```diff
@@ -132,6 +135,10 @@ export class DatePickerCalendar {
                 );
                 break;
         }
+
+        if (this.formatter) {
+            this.picker.formatDate = this.formatter;
```

Comment:
I really like this approach!

Right now `formatDate` is `protected` on the `Picker` class. We should either make it public or expose a setter method for the formatter?

Reply:
It seems like it's public, actually (I was a bit surprised about that)

Reply:
But I'm not sure if `picker.parseDate` still works 🤔

Reply:
It seems like it's not really needed when passing value as Date, so we should be fine leaving it unchanged.

Reply:
Ah, yeah it was actually public, my bad 🙈 
But anyhow, I still think it might be more explicit to set the formatter with a setter from the `Picker`. For someone reading the code it might be easier if we leave some breadcrumbs don't you think?